ipaserver/install/ipa_otptoken_import.py

```diff
@@ -506,6 +508,92 @@ def save(self, dest):
         self.__doc.write(dest)
 
 
+class CSVToken(object):
+    @property
+    def id(self):
+        return self.__id
+
+    @property
+    def options(self):
+        return self.__options
+
+    @property
+    def removed(self):
+        return self.__removed
+
+    def __init__(self, id, options):
+        self.__id = id
+        self.__options = options
+        self.__removed = False
+
+    def remove(self):
+        self.__removed = True
+
+
+class CSVDocument(object):
+    _FIELDS = ("serial", "key", "type", "digits", "interval")
+
+    _MAP = (
+        ('type', 'type', lambda v, o: v.strip()),
+        ('serial', 'ipatokenserial', lambda v, o: v.strip()),
+        (
+            'key',
+            'ipatokenotpkey',
+            lambda v, o: base64.b32encode(base64.b64decode(v)).decode('ascii')
+        ),
+        ('digits', 'ipatokenotpdigits', lambda v, o: v),
+        ('interval', 'ipatokentotptimestep', lambda v, o: v),
+    )
+
+    @property
+    def keyname(self):
+        # Always return None as we have no key names in CSV format
+        return None
+
+    def __init__(self, filename):
+        self.__keypackages = []
+
+        with open(filename, "r") as f:
+            reader = csv.reader(f)
+            for row in reader:
+                try:
```

Comment:
You may want to add a check on the number of items in each row (the code is expecting 5), otherwise it is possible to add a CSVToken that is missing some elements.